src/Console/InfectionContainer.php

```diff
@@ -232,12 +232,19 @@ public function __construct(array $values = [])
 
     public function buildDynamicDependencies(InputInterface $input): void
     {
-        $this['infection.config'] = function () use ($input): InfectionConfig {
+        $infectionConfigMaker = function () use ($input): InfectionConfig {
             $facade = new ConfigCreatorFacade($this['locator'], $this['filesystem']);
 
             return $facade->createConfig($input->getOption('configuration'));
         };
 
+        /*
+         * Configuration step needs a distinct instance of a configuration as it may be inexistent.
+         * Therefore we can avoid using this stale and invalid configuration later on.
+         */
+        $this['infection.config.initial'] = $infectionConfigMaker;
```

Comment:
I'm not sure that this is a good idea to have more than one config class.

Reply:
What makes you concerned?

Another option is to re-initialize this `infection.config` maker, which is a painful job with lots of moving parts.